### ISSUE TITLE
TypeError: cannot concatenate 'str' and 'NoneType' objects

### ISSUE THREAD

**Issue by adorton-adobe** [adobe-apiplatform/user-sync.py#68]
Customer is getting the following error:

```Traceback (most recent call last):
  File "c:\user-sync\.pex\install\user_sync-1.1.1-py2-none-any.whl.1d38cd7281dcf25d7c48135e8d87ce573a9f8c9f\user_sync-1.1.1-py2-none-any.whl\user_sync\app.py", line 270,
in main
    begin_work(config_loader)
  File "c:\user-sync\.pex\install\user_sync-1.1.1-py2-none-any.whl.1d38cd7281dcf25d7c48135e8d87ce573a9f8c9f\user_sync-1.1.1-py2-none-any.whl\user_sync\app.py", line 176,
in begin_work
    rule_processor.run(directory_groups, directory_connector, dashboard_connectors)
  File "c:\user-sync\.pex\install\user_sync-1.1.1-py2-none-any.whl.1d38cd7281dcf25d7c48135e8d87ce573a9f8c9f\user_sync-1.1.1-py2-none-any.whl\user_sync\rules.py", line 10
2, in run
    self.read_desired_user_groups(directory_groups, directory_connector)
  File "c:\user-sync\.pex\install\user_sync-1.1.1-py2-none-any.whl.1d38cd7281dcf25d7c48135e8d87ce573a9f8c9f\user_sync-1.1.1-py2-none-any.whl\user_sync\rules.py", line 16
2, in read_desired_user_groups
    user_key = RuleProcessor.get_directory_user_key(directory_user)
  File "c:\user-sync\.pex\install\user_sync-1.1.1-py2-none-any.whl.1d38cd7281dcf25d7c48135e8d87ce573a9f8c9f\user_sync-1.1.1-py2-none-any.whl\user_sync\rules.py", line 65
9, in get_directory_user_key
    return RuleProcessor.get_user_key(directory_user['username'], directory_user['domain'], directory_user['email'])
  File "c:\user-sync\.pex\install\user_sync-1.1.1-py2-none-any.whl.1d38cd7281dcf25d7c48135e8d87ce573a9f8c9f\user_sync-1.1.1-py2-none-any.whl\user_sync\rules.py", line 67
8, in get_user_key
    return username + ',' + domain
TypeError: cannot concatenate 'str' and 'NoneType' objects
2017-03-13 14:13:38 2448 INFO main - ========== End Run (Total time: 0:00:16) ===================
```

I don't want to include the whole log for the run because it contains PII stuff that we don't want in Github. 

It occurs after the directory user queries.  I haven't had time to do much investigation, but I current guess is that we're getting a malformed "mail" attribute that is causing problems with this function in `rules.py`:

```python
    @staticmethod
    def get_user_key(username, domain, email, isAdobeID=False):
        username = user_sync.helper.normalize_string(username)
        domain = user_sync.helper.normalize_string(domain)
        email = user_sync.helper.normalize_string(email)

        if isAdobeID:
            return '<AdobeID>' + email
        if (username == None):
            return email
        if (username.find('@') >= 0):
            return username
        return username + ',' + domain
```

The `user_email_format` is set to `{mail}`.  So the username should always be an email, and the condition to concatenate w/ the comma and domain should never be hit.  For some reason it is.

One thing to note is the customer is syncing one AD group containing two users.  Both users have a valid email address.  Are we pulling any data for users that don't belong to the sync group?

Is there anything we can do to anticipate this condition so the application doesn't crash?

I would like to see if I can reproduce this in my test environment.  I'll post additional information as I work on that.

**Comment by adobeDan** [comment]
@adorton-adobe This turns out to be a dup of one of the issues I found and fixed with #47.  Can you try reproing in the tip of the v1 branch?  I think it will be gone.

**Comment by adobeDan** [comment]
Yes, I've confirmed this is a duplicate of #25, which is fixed in v1.  Closing as a dup.
